webpack/assets/javascripts/react_app/routes/index.js

```diff
@@ -1,11 +1,26 @@
-import React from 'react';
-import { Switch, Route } from 'react-router-dom';
+import React, { useEffect } from 'react';
+import { Switch, Route, useLocation } from 'react-router-dom';
 import URI from 'urijs';
 import { routes } from './routes';
 
 let currentPath = null;
 
 const AppSwitcher = () => {
+  const location = useLocation();
+
+  useEffect(() => {
+    const nextPath = getURIPath();
+    if (currentPath !== nextPath) {
+      updateCurrentPath();
+      if (!isRegisteredRoute(nextPath)) {
+        handleTurbolinksVisit(location, nextPath);
+      }
+    }
+  }, [location]);
+
+  const isRegisteredRoute = nextPath =>
+    routes.includes(({ path }) => path === nextPath);
```

Comment:
Would it work with routes with variables?
like in foreman-tasks:
- `/foreman_tasks/tasks/:id/sub_tasks`
- `/foreman_tasks/ex_tasks/:id`

Reply:
True, it won't work.
I will try to use [matchPath](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/matchPath.md) or something similar that react router uses too.